src/components/TextInput/TextInput.stories.tsx

```diff
@@ -150,4 +151,32 @@ export const BadAccessibility = () => {
   return <TextInput />;
 };
 
+export const TypeSwitch = () => {
+  const [type, setType] = useState("password");
+  const toggle = useCallback(() => {
+    if (type === "password") {
+      setType("text");
+    } else {
+      setType("password");
+    }
+  }, [type]);
```

Comment:
nit: I'd pass the `type` as a prop instead of having the method regenerate on `type` change.
```suggestion
  const toggle = useCallback((type) => {
    if (type === "password") {
      setType("text");
    } else {
      setType("password");
    }
  }, []);
  // other stuff
  <Checkbox
        onChange={() => toggle(type)}
```

However, as this is such a small component, and it doesn't have any children, this suggested change would have zero actual impact, so it doesn't really matter. So feel free to ignore. But I did want to point it out as this would improve the efficiency of the code (as far as I can tell).

Reply:
I'll ignore it as it is minor and is in the story file.